mvc/src/test/java/com/interface21/webmvc/servlet/mvc/tobe/HandlerExecutionsTest.java

```diff
@@ -0,0 +1,34 @@
+package com.interface21.webmvc.servlet.mvc.tobe;
+
+import com.interface21.web.bind.annotation.RequestMapping;
+import com.interface21.web.bind.annotation.RequestMethod;
+import jakarta.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.Test;
+import samples.TestController;
+
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class HandlerExecutionsTest {
+
+    private final HandlerExecutions handlerExecutions = new HandlerExecutions();
+
+    @Test
+    void mappingHandler() {
+        final Method method = mock(Method.class);
+        final RequestMapping requestMapping = mock(RequestMapping.class);
```

Comment:
`method`와 `requestMapping`은 mock을 사용하지 않을 수 있을 것 같은데, 어떻게 생각하실까요??

Reply:
Method의 경우에는 생성자가 defualt이고(+ Method를 만들기 위해 부가적으로 필요한 객체가 너무 많아요...), RequestMapping 의 경우 어노테이션이라 객체를 만들어 사용하기 어려운 것 같아요.😢 

혹시 mock을 사용하지 않는 것이 실제 객체를 사용하라는 의미가 아니라면 낙낙이 생각하신 방법에 대해 공유해주실 수 있으신가요?🥺

Reply:
```suggestion
        final Method method = TestController.class.getMethod("findUserId", HttpServletRequest.class, HttpServletResponse.class);
        final RequestMapping requestMapping = method.getAnnotation(RequestMapping.class);
```
아하!! 위와같이 `TestController.class` 로부터`Method`와 `RequestMapping` 을 실제로 받아와서 사용해보면 어떨지 여쭈어 본 것이었습니다!

조금 더 상세하게 리뷰 남겨 드릴 걸 그랬네요 ㅎㅎ

Reply:
그 말씀이었군요 !! 설명해주셔서 감사합니다 😁

제가 테스트를 하고 싶은 부분은 **`@RequestMapping`의 url과 method에 따라 특정 핸들러가 매핑되었는지**를 확인하고 싶었어요. 그래서 `@RequestMapping`이 어떤 값을 가졌는지 테스트에 잘 드러났으면 좋겠어요.

```java
    when(requestMapping.value()).thenReturn("/get-test");
    when(requestMapping.method()).thenReturn(new RequestMethod[]{RequestMethod.GET});
``` 
이 부분에서 매핑을 원하는 값이 잘 드러난다고 생각해요.

낙낙의 방식은 실제 구현을 밀접하게 테스트 가능한 장점이 있어요. 
그럼에도 제 입장에서는 `/get-test`와 `RequestMethod.GET`로 매핑한다는 흐름이 중요한 것 같아요. TestController에서 메서드를 리플렉션으로 가져오는 방식은 어떤 url과 method를 가졌는지 한 눈에 확인하기는 어려운 것 같아 현재 방법을 유지하고자 합니다. 낙낙의 의견은 어떤지 궁금합니다 !

Reply:
아하!! 확실히 `TestController.class`에 들어가지 않으면 어떤 url과 method에 따라 매핑되는지 테스트 코드에서 바로 보이지 않는 문제가 있겠네요!

그렇다면 저는 테니의 방식도 좋다고 생각합니닷!! 그대로 킵 고잉 하시죠 ㅎㅎ